libhwmon/wrapadl.cpp

```diff
@@ -188,6 +206,16 @@ int wrap_adl_get_fanpcnt(wrap_adl_handle *adlh, int gpuindex, unsigned int *fanp
 	return 0;
 }
 
+int wrap_adl_get_power_usage(wrap_adl_handle *adlh, int gpuindex, unsigned int *power)
+{
+	wrap_adlReturn_t rc;
+	if (gpuindex < 0 || gpuindex >= adlh->adl_gpucount){
+		return -1;
+	}
+	rc = adlh->adl2Overdrive6CurrentPowerGet(adlh->context, adlh->phys_logi_device_id[gpuindex], 0, (int*)power);
+	return rc;
+}
+
```

Comment:
please consider that this function return milliwatt directly (like the other functions).
This way we have a consistent interface and do not need to convert the result differently per hw.

Reply:
I'll change it on next commit. I thought about it, actually, but I had an unexpected issue with that, and as was a bit late, I let it as it worked.